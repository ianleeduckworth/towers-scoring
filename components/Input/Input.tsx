import React from 'react';
import {
  NativeSyntheticEvent,
  StyleSheet,
  TextInput,
  TextInputFocusEventData,
  View,
  Text,
} from 'react-native';
import { defaults } from '../../styles/global';

interface InputProps {
    type?: 'text' | 'numeric',
    label: string;
    placeholder?: string;
    onChangeText?: ((text: string) => void) | undefined;
    value?: string;
    onBlur?: ((e: NativeSyntheticEvent<TextInputFocusEventData>) => void) | undefined;
    errorText?: string;
}

export const Input = ({
  type = 'text',
  label,
  placeholder,
  onChangeText,
  value,
  onBlur,
  errorText,
}: InputProps) => (
        <View style={styles.container}>
            <Text style={styles.label}>{label}</Text>
            <TextInput
                keyboardType={type === 'numeric' ? 'number-pad' : 'default'}
                style={styles.input}
                placeholder={placeholder}
                onChangeText={onChangeText}
                value={value}
                onBlur={onBlur}
            />
            <Text style={styles.errorText}>{errorText}</Text>
        </View>
);

const styles = StyleSheet.create({
  container: {
    minWidth: 360,
  },
  errorText: {
    color: defaults.red,
    fontSize: defaults.normalTextSize,
    fontWeight: 'bold',
    marginBottom: 10,
    marginTop: 6,
  },
  input: {
    borderColor: defaults.mediumGray,
    borderRadius: 5,
    borderWidth: 1,
    fontSize: defaults.normalTextSize,
    padding: 10,
  },
  label: {
    color: defaults.black,
    fontSize: defaults.normalTextSize,
    fontWeight: 'bold',
    marginBottom: 5,
  },
});
