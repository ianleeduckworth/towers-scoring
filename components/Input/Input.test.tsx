import renderer from 'react-test-renderer';
import * as React from 'react';
import { Input } from './Input';

describe('card', () => {
  it('base renders correctly', () => {
    const element = renderer.create(
            <Input label="Test Input" />,
    ).toJSON();

    expect(element).toMatchSnapshot();
  });

  it('renders correctly with placeholder', () => {
    const element = renderer.create(
            <Input
                label="Test Input"
                placeholder="Test placeholder"
            />,
    );

    expect(element).toMatchSnapshot();
  });

  it('renders correctly with error text', () => {
    const element = renderer.create(
            <Input
                label="Test Input"
                placeholder="Test placeholder"
                errorText="Test error text"
            />,
    );

    expect(element).toMatchSnapshot();
  });

  it('renders correctly with value', () => {
    const element = renderer.create(
            <Input
                label="Test Input"
                placeholder="Test placeholder"
                errorText="Test error text"
                value="Value"
            />,
    );

    expect(element).toMatchSnapshot();
  });
});
