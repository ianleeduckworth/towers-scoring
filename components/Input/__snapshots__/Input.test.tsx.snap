// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card base renders correctly 1`] = `
<View
  style={
    Object {
      "minWidth": 360,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#000000",
        "fontSize": 18,
        "fontWeight": "bold",
        "marginBottom": 5,
      }
    }
  >
    Test Input
  </Text>
  <TextInput
    allowFontScaling={true}
    keyboardType="default"
    rejectResponderTermination={true}
    style={
      Object {
        "borderColor": "#F0F0F0",
        "borderRadius": 5,
        "borderWidth": 1,
        "fontSize": 18,
        "padding": 10,
      }
    }
    underlineColorAndroid="transparent"
  />
  <Text
    style={
      Object {
        "color": "#B80000",
        "fontSize": 18,
        "fontWeight": "bold",
        "marginBottom": 10,
        "marginTop": 6,
      }
    }
  />
</View>
`;

exports[`card renders correctly with error text 1`] = `
<View
  style={
    Object {
      "minWidth": 360,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#000000",
        "fontSize": 18,
        "fontWeight": "bold",
        "marginBottom": 5,
      }
    }
  >
    Test Input
  </Text>
  <TextInput
    allowFontScaling={true}
    keyboardType="default"
    placeholder="Test placeholder"
    rejectResponderTermination={true}
    style={
      Object {
        "borderColor": "#F0F0F0",
        "borderRadius": 5,
        "borderWidth": 1,
        "fontSize": 18,
        "padding": 10,
      }
    }
    underlineColorAndroid="transparent"
  />
  <Text
    style={
      Object {
        "color": "#B80000",
        "fontSize": 18,
        "fontWeight": "bold",
        "marginBottom": 10,
        "marginTop": 6,
      }
    }
  >
    Test error text
  </Text>
</View>
`;

exports[`card renders correctly with placeholder 1`] = `
<View
  style={
    Object {
      "minWidth": 360,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#000000",
        "fontSize": 18,
        "fontWeight": "bold",
        "marginBottom": 5,
      }
    }
  >
    Test Input
  </Text>
  <TextInput
    allowFontScaling={true}
    keyboardType="default"
    placeholder="Test placeholder"
    rejectResponderTermination={true}
    style={
      Object {
        "borderColor": "#F0F0F0",
        "borderRadius": 5,
        "borderWidth": 1,
        "fontSize": 18,
        "padding": 10,
      }
    }
    underlineColorAndroid="transparent"
  />
  <Text
    style={
      Object {
        "color": "#B80000",
        "fontSize": 18,
        "fontWeight": "bold",
        "marginBottom": 10,
        "marginTop": 6,
      }
    }
  />
</View>
`;

exports[`card renders correctly with value 1`] = `
<View
  style={
    Object {
      "minWidth": 360,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#000000",
        "fontSize": 18,
        "fontWeight": "bold",
        "marginBottom": 5,
      }
    }
  >
    Test Input
  </Text>
  <TextInput
    allowFontScaling={true}
    keyboardType="default"
    placeholder="Test placeholder"
    rejectResponderTermination={true}
    style={
      Object {
        "borderColor": "#F0F0F0",
        "borderRadius": 5,
        "borderWidth": 1,
        "fontSize": 18,
        "padding": 10,
      }
    }
    underlineColorAndroid="transparent"
    value="Value"
  />
  <Text
    style={
      Object {
        "color": "#B80000",
        "fontSize": 18,
        "fontWeight": "bold",
        "marginBottom": 10,
        "marginTop": 6,
      }
    }
  >
    Test error text
  </Text>
</View>
`;
